### ISSUE TITLE
Fifa 21 GPU hang with latest proton-tkg

### ISSUE THREAD

**Issue by yoyossef** [Frogging-Family/wine-tkg-git#280]
Hello,

I'm getting an issue with Fifa 21 on Steam using proton-tkg, the game runs fine up to the build I've made the 25th of January at 08:25 PM CEST.

My latest "good" build:

`-rw-r--r-- 1 user wheel 283125349 Jan 25 20:15 proton-tkg-git-6.0.r12.g3dec70bf-1-x86_64.pkg.tar.zst`

But when I rebuilt proton-tkg 3 days later at 05:13 PM or even the build of today, I'm getting a GPU hang on the game's loading screen. The screen gets black, gnome shell gets all weird and I'm forced to reboot the computer or hard shutdown it.

A screenshot of what I see just before the GPU hang, can't even get to the language selector screen (took it off of YouTube to save time, couldn't get a screenshot before the hang).

![image](https://user-images.githubusercontent.com/31565280/106355008-daa9c200-62f5-11eb-9a91-b618f50f8f2d.png)

The faulty builds:
```
-rw-r--r-- 1 user wheel 283201501 Jan 28 17:13 proton-tkg-git-6.0.r16.g9692b2e5-1-x86_64.pkg.tar.zst
-rw-r--r-- 1 user wheel 283041322 Jan 30 12:05 proton-tkg-git-6.0.r18.g5c5a8f3b-1-x86_64.pkg.tar.zst
```

So my workaround is to stick with the old working build.

[System Information](https://gist.github.com/yoyossef/35c0a46c0513059c7267e6df182ea369)

**Comment by CodingKoopa** [comment]
Could you see if there are any relevant errors in your system log when it happens by running `journalctl -e`? Or, if your system has locked up, then on the next boot with `journalctl -b -1 e`?

**Comment by yoyossef** [comment]
> Could you see if there are any relevant errors in your system log when it happens by running `journalctl -e`? Or, if your system has locked up, then on the next boot with `journalctl -b -1 e`?

@CodingKoopa [journalctl logs](https://gist.github.com/yoyossef/28cbd5375875bcde01e08546d5472633)

I guess there are some relevant parts, they show in red in the terminal and they're about AMDGPU.

**Comment by Tk-Glitch** [comment]
That tends to point to DXVK being the culprit. There has been 6 commits on the 28th of january https://github.com/doitsujin/dxvk/commits/master

**Comment by yoyossef** [comment]
> That tends to point to DXVK being the culprit. There has been 6 commits on the 28th of january [doitsujin/dxvk@`master` (commits)](https://github.com/doitsujin/dxvk/commits/master)

@Tk-Glitch I used dxvk-tools to build specific builds of DXVK for proton-tkg (different commits from january 28th to assess where exactly the issue came from), notably https://github.com/doitsujin/dxvk/commit/1478011822435e5f52addebf4e1c0b6e4bc6c6bc, https://github.com/doitsujin/dxvk/commit/f5fa7a909961692431306f0c07f583a54cfece49 and https://github.com/doitsujin/dxvk/commit/7d7c057d468256eb92c0ec81d3d9593512f90f5f but I'm still getting the same freeze at the same moment.

I'm still investigating on the DXVK part and will update the post as I test older commits.

**Comment by yoyossef** [comment]
Okay so I found what the issue was, for some reason the older build would ignore the config file indicating that the game should use DX12 instead of DX11.

So the GPU hang is VKD3D-Proton related I think, when I switched back to DX11/DXVK the game runs just fine.

Make sure that `fifasetup.ini` in `/compatdata/1313860/pfx/drive_c/users/steamuser/My Documents/FIFA 21` has the value `DIRECTX_SELECT` set to 1 to use DX11 (value 0 to test DX12).
